Delphi.Mocks.Helpers.pas

```diff
@@ -102,6 +104,43 @@ function CompareValue(const Left, Right: TValue): Integer;
   end;
 end;
 
+function SameVariant(const Left, Right: Variant): Boolean;
+var
+  i: integer;
+begin
+  // here, we are going to check to the best of our ability that the variants are the same.
+  // simply a = b is not enough in some cases with variants
+
+  // a varArray of variants needs to be correctly checked, as implicit comparison does not work
+  if ( VarType(Left) = VarType(Right) ) AND
+     ( VarType(Left) = varArray + varVariant ) then
+  begin
+    // first check if the pointers are the same, which would indicate the same variant instance
+    Result := @Left = @Right;
```

Comment:
I tried this in my implementation and this check was never true. Also I don't know if that would actually matter. Also from what I can see the test that should check this is defect as it does not test for the same instance - its exactly the same as the test for different instances.